File: core/frontend/src/types/comment.ts
export interface comment {
  id: number,
  text: string,
  published_at: string,
  likes: number,
  comments_count: number,
  answers: answer[],
  isShowedAnswers: boolean,
  isLiked: boolean,
  user: {
    username: string,
    avatar: string
  }
}

export interface answer {
  id: number,
  text: string,
  published_at: string,
  likes: number,
  answer_to: number,
  isLiked: boolean,
  user: {
    username: string,
    avatar: string
  }
}

File: core/backend/Models/Comment.py
from django.db import models
import datetime

from .Post import Post
from .User import User

class Comment(models.Model):
  post = models.ForeignKey(Post, blank=True, null=True, on_delete=models.CASCADE)
  user = models.ForeignKey(User, blank=True, null=True, on_delete=models.CASCADE)
  answer_to = models.ForeignKey("self", blank=True, null=True, on_delete=models.CASCADE)
  text = models.TextField()
  published_at = models.DateTimeField(default=datetime.datetime.now)
  likes = models.IntegerField(default=0)
  likers = models.ManyToManyField(User, blank=True, related_name="users who liked this comment+")
  
  def __str__(self):
    return self.text